src/js/components/modals/JobFormModal.js

```diff
@@ -62,7 +63,8 @@ class JobFormModal extends mixin(StoreMixin) {
       defaultTab: '',
       errorMessage: null,
       job: new Job(),
-      jsonMode: false
+      jsonMode: false,
+      jsonDefinition: ''
```

Comment:
Let's please not have more than one state property to store the job definition, as this makes it really hard to reason about state changes and might result in a wrong state.